packages/flame/lib/src/collisions/hitboxes/shape_hitbox.dart

```diff
@@ -89,10 +89,11 @@ mixin ShapeHitbox on ShapeComponent implements Hitbox<ShapeHitbox> {
       _validAabb = false;
       onAabbChanged?.call();
     };
-    ancestors(includeSelf: true).whereType<PositionComponent>().forEach((c) {
-      _transformAncestors.add(c.transform);
-      c.transform.addListener(_transformListener);
-    });
+    for (final ancestor
+        in ancestors(includeSelf: true).whereType<PositionComponent>()) {
```

Comment:
Nit
```suggestion
    fina positionComponents = ancestors(includeSelf: true).whereType<PositionComponent>();
    for (final ancestor in positionComponents) {
```

Reply:
Hehe, you missed an `l` in final :smile:

Reply:
Tried my best without a compiler analyzer at my disposal :P